yelp_bytes.py

```diff
@@ -26,22 +37,24 @@ def from_bytes(obj, encoding='internet', errors='strict'):
 
     This function always returns unicode.
     """
-    if isinstance(obj, unicode):
+    mro = type(obj).mro()
+    if unicode in mro:
         return obj
+    elif bytes in mro:
+        return obj.decode(encoding, errors)
+    elif hasattr(obj, '__bytes__'):
+        return obj.__bytes__().decode(encoding, errors)
```

Comment:
the tests fail because the tests are written for py2 semantics: `unicode()` will try `__unicode__`, and if it doesn't exist (which it often doesn't), will fall back to calling `__str__` and decoding the result with the default encoding

this is no longer the case in py3, where the more commonly-implemented `__str__` is now the text type and the workaround isn't necessary.  so your classes that implement `__bytes__` but not `__str__` stringify to the default representation, as they should.  that's why i skipped these tests on py3.

i interpret the goal of this function to be "get me some text", not "dig up some bytes and then try to decode them"  :(